### ISSUE TITLE
Hint previews sooner

### ISSUE THREAD

**Issue by BadMagic100** [ArchipelagoMW-HollowKnight/Archipelago.HollowKnight#152]
Today, previewable locations hint automatically when you leave the room. This makes some amount of sense, if you see a thing in a shop and immediately buy the thing, you probably don't want it hinted, and shop items do have some indicator of the amount of value, indicating whether they are progression, useful, or not useful.

However, this doesn't account for the fact that progression/useful items may vary in terms of cost vs benefit *right now*, and some progression/useful items may never be worth the price. In addition, not all previewable locations are shops and may not have this information available on a surface level. What happens in my experience is that players are still asking "is X useful" or reloading the room to hint and then say "are any of those things useful."

Let's just make this easier on players and pull the hints forward to when the IC event fires, knowing that it may create some hints for things which will be bought immediately

**Comment by BadMagic100** [comment]
Actually let's make this a mod option, defaulting to the new behavior. Players can choose to revert to the current behavior, or to not create any hints at all

**Comment by KonoTyran** [comment]
When you say mod option. How do you invision this? As an option on the yaml? Or do we make a config menu option on the pause mod options menu? Or some other method?

**Comment by BadMagic100** [comment]
Yeah, I'm thinking mod menu option. Probably IMenuMod (no satchel) since it's just a simple toggle. I don't believe it's something that belongs in the yaml because it doesn't affect gameplay that much - like, if someone turned it off for a race it wouldn't really matter if you know what you're doing

**Comment by BadMagic100** [comment]
I now believe there is no need for an option controlling this, as hint priority is merged and we can auto hint at low priority now

**Comment by BadMagic100** [comment]
After checking on hint prio details, this can be implemented now, as create_as_hint hints have unspecified priority. But we could also keep a watch on the following pr for a dedicated CreateHints packet as we get no value out of the scouts right now other than the hint https://github.com/ArchipelagoMW/Archipelago/pull/4317
